### ISSUE TITLE
OsmAnd is slow on Android O [Fixed in Android]

### ISSUE THREAD

**Issue by liondog** [osmandapp/Osmand#4323]
Just updated a Nexus 6P to Android O and reinstalled all apps including OsmAnd+ in the current version v2.7.5. The app shows up and GUI elements just work as inteded. However, when interacting with the map itself the app becomes unbearable slow and unresponsive. Especially zooming is extremly slow. Eventually the map reacts and the app doesn't freeze/crash. But a normal usage is not possible under this condition.

**Comment by vshcherb** [comment]
We still need to test on real device, so far emulator didn't show any bad signs.

**Comment by rocketsound** [comment]
Can I support you in testing this? Do you just need "real evidence" on the matter or do you need more details? However, I don't have any experience in Android app programming.

**Comment by thestinger** [comment]
This also occurs on a Pixel XL with both stock Android and a build of the Android Open Source Project.

A potential issue in the logs:

```
08-24 23:10:01.128  3677  3701 D net.osmand: NativeOsmandLibrary Loading jnigraphics, since Android >= 2.2 ...
08-24 23:10:01.128  3677  3701 W net.osmand.plus: 0x8777e438
08-24 23:10:01.129  3677  3701 D net.osmand: NativeOsmandLibrary Failed to load jnigraphics: java.lang.UnsatisfiedLinkError: Shared library "/system/lib/libjnigraphics.so" already opened by ClassLoader 0x0; can't open in ClassLoader 0x8777e494
```

Mentioned again:

```
08-24 23:10:02.761  3677  3699 W net.osmand:native: jnigraphics was not found in loaded libraries
08-24 23:10:02.768  3677  3699 W net.osmand:native: jnigraphics was not loaded in default location
```

Odd output:

```
08-24 23:10:02.882  3677  3699 I net.osmand:native: Native ok (rendering -205797600, text -1535657454 ms) 
08-24 23:10:02.882  3677  3699 I net.osmand:native:  (98608 points, 27 points inside, 38 of 67 objects visible)
08-24 23:10:02.882  3677  3699 I net.osmand:native: End Rendering image
08-24 23:10:02.882  3677  3699 I net.osmand:native: Native ok (init -206017190, rendering -205797600) 
```

It lags especially when zooming, which is logged:

```
08-24 23:11:39.962  3768  3768 I Choreographer: Skipped 59 frames!  The application may be doing too much work on its main thread.
08-24 23:11:40.958  3768  3768 I Choreographer: Skipped 58 frames!  The application may be doing too much work on its main thread.
08-24 23:11:41.966  3768  3768 I Choreographer: Skipped 60 frames!  The application may be doing too much work on its main thread.
```

I can get more complete logs but there was some personal information in the output and this might be enough to resolve it.

**Comment by kshpytsya** [comment]
This is just a "me too" on Pixel XL. Willing to assist with investigation.

**Comment by vshcherb** [comment]
The investigation is almost finished but we don't know the fix yet. The problem in the jnigraphics.so 
System.loadLibrary("jnigraphics");

It was included since Android 2.2 and that library OsmAnd relied. There were no announcement that is going to be deprecated and it is not clear what to do. Basically that library load is not supported any more.

This library is important cause it provides API to work with Android images on low level.

**Comment by thestinger** [comment]
I don't think it was ever supported to use the system copy via loadLibrary, it needs to be added to an NDK project with `LOCAL_LDLIBS += -ljnigraphics`.

**Comment by kshpytsya** [comment]
Do I understand it correctly that the solution for the problem is found but nothing is done for three days? As a paying customer (which I am) I would expect you to roll a hotfix (whatever you call it) -- ASAP -- which would be based on your current release and fix just this issue instead of (correct me if I am wrong) planning to include the fix in your next release (isn't it what "vshcherb added this to the 2.8 milestone 13 days ago" means?) This is a critical showstopper for me: I am essentially unable to use the software for which I paid and on which I rely.

**Comment by vshcherb** [comment]
@kshpytsya you can also request a hotfix from Android O or a device issue. Why do you think it is an OsmAnd issue?

**Comment by vshcherb** [comment]
@thestinger  https://github.com/osmandapp/OsmAnd-core/commit/63d83eb31219daa60e8abeacfbb060567b810d10 looks like didn't help...

**Comment by vshcherb** [comment]
The fix will be more complicated we are going to change from dynamic linking to static linking that library, it will require testing on multiple platforms. The link to a test build will be provided.

**Comment by BalooUriza** [comment]
Issue is reproducible on OsmAnd~ 2.7.5 on Android 8.0.0 running on a Pixel XL.

**Comment by vshcherb** [comment]
There is a new nightly build where the native library is fixed and the rendering itself is faster. Though it seems it is not the only issue, we are try to find what has changed. It looks like the Canvas API may be changed or there is a limitation on concurrent threads.

Unfortunately there is not enough information in the internet yet.

**Comment by vshcherb** [comment]
The fix No 1 that was just merged https://github.com/osmandapp/OsmAnd-core/pull/66/files.

**Comment by vshcherb** [comment]
That's possibly an android bug https://issuetracker.google.com/issues/64882137. So far identified that drawBitmap takes unexplainably too much time here https://github.com/osmandapp/Osmand/blob/master/OsmAnd/src/net/osmand/plus/views/OsmandMapTileView.java#L531

**Comment by vshcherb** [comment]
To keep updated:
1) Performance issues fixed: native library is supported & panning map works smoothly
2) Zooming fps is not fixed: it works very slow & the last method we traced was https://github.com/osmandapp/Osmand/blob/master/OsmAnd/src/net/osmand/plus/render/MapVectorLayer.java#L194. It sounds like this open issue https://issuetracker.google.com/issues/64882137.

So far no workaround found (2) & we are working to get simple example with this regression. Anyway nightly build should work much better already than 2.7.5

**Comment by rbovard** [comment]
Thanks for the update @vshcherb.

Do you plan to release a 2.7.6 version with the first issue adressed?

**Comment by vshcherb** [comment]
The release is planned in the beginning of October and we still rely on having 2 more weeks for final fix & more testing. 2.7.6 is not feasible to prepare in parallel, so release will be current master branch 2.8

**Comment by scaidermern** [comment]
The underlying issue in Android has been fixed: https://issuetracker.google.com/issues/64882137

This will require an Android update, though.

**Comment by vshcherb** [comment]
Sounds good, anyway we are planning to have beta build where the situation has improved the only thing it is laggish during zoom in.

**Comment by vshcherb** [comment]
Major fixes were done in 2.8. 
1. First of all, now native library could be used & the routing is fast again.
2. We are waiting for new Android Update which will fix the issue with slowness
If it doesn't happen (Android update), we plan to create new issue for Android.

**Comment by vshcherb** [comment]
Latest test on Android 8.0:
1. The rendering / routing speed the same as on 7.0. Full screen renders in 0.5-1.5 seconds.
2. FPS is still lower and causes some bad effects. FPS is at level 10, where good / instant level is 60.

**Comment by valynor** [comment]
Just installed 2.7.5 on my Nexus 6P (stock Oreo 8.0.0) and I don't notice any improvement in panning the map - it's still super sluggish. Zooming in/out is worse, though.

**Comment by naoliv** [comment]
@valynor it seems that you need osmand 2.8.0+

**Comment by valynor** [comment]
I'm sorry, I confused the version numbers. I have 2.8.2 installed.

**Comment by rbovard** [comment]
It's the same here with 2.8.2, I think we should wait on the Android update.

**Comment by vshcherb** [comment]
I think it is not the same just because before 2.8, OsmAnd was running only in Safe mode (you could check it in General Settings). So the difference definitely should exist

**Comment by brncsk** [comment]
Rendering while panning have certainly become faster in recent nightlies, but it still lags badly when zooming.

I realized, however, that QT nightlies don't exhibit these problems. Caveat, though: they have a weird regression that prevents you from single-tapping POIs to reveal their property sheets.

**Comment by rbovard** [comment]
FYI I just get the November Android update and it's still laggy (with 2.8.2).

**Comment by lebenlechzer** [comment]
Same for me, no improvements visible after the November update.

**Comment by raindropsfromsky** [comment]
I have not observed any speed issues on OnePlus 3T after update to Android 8.0.0 (OxygenOS 5.0).
OsmAnd~ 3.0.0#23181M (released on 20th Nov 2017)

**Comment by pkasperskyi** [comment]
Just performed Android upgrade to 8.1 BETA DP2 on Nexus 5X.

Now OsmAnd visual performance is better. Test performed for both 2.8.2 available in Play Market and today's Nightly Build.

Some lagging still remains during redraw on zoom in/out and on panning. I can't say for sure if it corresponds to the performance on Android 7, but at least application is usable again.

**Comment by vshcherb** [comment]
*Important*: some issues are already fixed but not all of them. Mostly it's visible during zoom in/out.
As 28 November 2017 OsmAnd 2.8.

**Comment by JKAbrams** [comment]
@vshcherb: is more debugging help needed in order to better understand those remaining issues or are you talking about the issues that by the above reports seem to be fixed in the upcoming Android 8.1 release due in December?

**Comment by vshcherb** [comment]
Latest update: we don't see any issue with map panning in Android 8.1 beta 2, though we still see issues with map zoom in. It looks like Android team fixed the issue partially we are preparing detailed description to reopen a bug next week.

**Comment by jypma** [comment]
A seemingly follow-up google bug at https://issuetracker.google.com/issues/70113293 has yesterday been reported "fixed". Unclear what Android release that'll be in though.

**Comment by vshcherb** [comment]
This is great, we are looking forward with it.

**Comment by valynor** [comment]
I can offer some first-hand insights into Oreo 8.1 performance, got the update yesterday (Nexus 6P):
-panning is now as fast as before. the difference is huge an immediately noticeable 
-zooming is a little faster too but still semi-sluggish

At least Osmand is usable again (for me).

**Comment by JKAbrams** [comment]
First run took about 2 minutes to start. Panning might be slightly faster but honestly I cannot rellay see any difference, 'common worst case' is about 5 seconds to render the screen. The big difference for me with panning speed was the upgrade to Osmand 2.8.2. Zooming seems more responseive to me, it is faster before _something_ is displayed, which gives you a chance to react better. Selecting safe mode now gives error messages "Error ocurred while rendering selected area" when panning.
Android 8.1 on CopperheadOS / Nexus 5x.

**Comment by riggs-** [comment]
Most device vendors are rolling out upgrades to Oreo 8.0 first, before (hopefully) moving to 8.1. This means most "up-to-date" non-Google-devices will run a version Oreo suffering from this issue. Is there any chance of this getting fixed for Oreo 8.0?

**Comment by erikrr** [comment]
Here also an terribly slow Oreo/Osmamd victim  :( 
5 second waiting time when i zoom in or out. I unchecked "high detail"  its still very very slow but at least its workable.

Tried some thing in developer settings as opengl rendering, both Standard and Skia but didnt help.
 
Oneplus 5 user with the newest version of OSMAMD (2.82).
Android 7 was a speedy expercience but with V8 end december its almost impossible to work with. 
Hoping for a quick solution because i ravel a lot to new city's so i need an good gps/map program and i hade one ...

**Comment by riggs-** [comment]
Yes, even in browsers (e.g. Firefox mobile) pan and zoom on openstreetmap.org works like a charm. How do they get around this problem?

**Comment by vshcherb** [comment]
As I see latest update has fixed that issue but it might no be available on all devices yet.

**Comment by Apolyxon** [comment]
So is this fixed for Version 2.9? This version has been released on the Google Play Store already? .. Is there a way to sideload it? I am on the OsmAnd Beta - yet I still don't have this version.

**Comment by vshcherb** [comment]
There will be no fixes provided in APK, so 2.8 will be functioning fine once Android O is updated.

**Comment by Apolyxon** [comment]
So you are telling me I have to downgrade to Android 7? Android 8.1 is months away for Samsung Phones.

**Comment by JKAbrams** [comment]
Has the fix been released by Google? What's the exact Android version we are looking for?

**Comment by thestinger** [comment]
8.1 (mr1 branch)

It's still pretty slow either way... it was pretty slow before this regression too, but it's worse than it was.

**Comment by Apolyxon** [comment]
If you seriously decide that waiting for 8.1 is a viable option, let warn you: Samsung is about to update all their newest devices to Oreo 8.0 - as well as all the other manufacturers.

OsmAnd has taken a hit with Play Store ratings already but that was nothing compared to what is going to come in the next months for this app. The only people who are on Oreo are Pixel/Nexus owners and some rare early adopters like me on the Samsung Oreo Beta.

Oreo 8.0 is on 0.5 % of all devices as of January, 8. Oreo 8.1 is on 0.2 % of devices.

This is going to blow up very rapidly, very soon.

**Comment by jypma** [comment]
Why are other map apps (apparently) still smooth in 8.1? Are they using OpenGL instead of `Canvas`?

**Comment by utack** [comment]
agree with @Apolyxon , this should be moved to the 2.9 milestone
I get why you are annoyed Google broke it without warning, but Osmand should not be the app eating all the bad ratings for this

**Comment by vshcherb** [comment]
We have some devices already on Android 8.0 where that issue has been fixed and OsmAnd works at the same speed as older version.
Again there is not much to fix in OsmAnd cause it is using standard Canvas API, there is no another API method to call or at least we don't know.

**Comment by Apolyxon** [comment]
So the Bugfix is not only in Android 8.1 but also in a Bugfix version for Android 8.0? Probably the January version?

That is a rather important distinction: Most people on Oreo 8.0 would get such a bugfix quite quickly. Android 8.1 - not so much.

**Comment by vshcherb** [comment]
I think it would worth to post some videos to be sure we are talking exactly same issue. Probably some reporters here confuse "slowness" to Google Maps with actual regression on Android 8.0. 

I observed that regression myself in 2 situations on Google Nexus (Model?) : 1) when panning the map (already fixed) 2) when zooming out/in (not fixed)

On another phone - OnePlus Android 8.0 Oreo - both issue are already fixed

It is easy to check the slowness with FPS widget, it actually displays number < 10-20 where on average it should be 50.
![image](https://user-images.githubusercontent.com/1042025/35361471-bdbafaf4-0161-11e8-93df-19650d1fdd76.png)

**Comment by Apolyxon** [comment]
@vshcherb 

I am pretty sure we are talking about the same thing here. The zooming performance is abhorrently bad.

I made a video showing this. When zooming I get maybe 1-3 fps. When the app finally finished the zooming process, panning around is very smooth and normal.

This is recorded on a Samsung Galaxy S8 with latest Oreo Beta 8.0 and December Security Patch level.

![ezgif-1-cf2165d152](https://user-images.githubusercontent.com/3678079/35383103-9443085a-01c1-11e8-8a72-804e1113148c.gif)

**Comment by Apolyxon** [comment]
@vshcherb 
Awesome, thank you. I will report back when the official Oreo release happens for the S8.

**Comment by riggs-** [comment]
I would be very surprised if this was fixed in the official Oreo release for the S8. I have tested this on a HTC U11 with the official Oreo 8.0 release and it looks exactly like @Apolyxon 's screen recording. And once you zoom in to a detailed level with contours and hillshades, it *is* slow.

**Comment by erikrr** [comment]
Today in noticed that everything is working normal again, or at least it seems normal. Speed and reaction speed is ok now on my Oneplus 5. I guess the fix was somewhere in de OnePlus monthly update a few days back.
Great news!

**Comment by Apolyxon** [comment]
@vshcherb 

I have just updated my Samsung Galaxy S8 to the official Oreo release: G950FXXU1CRAP. This is the February Patch Level.

I can confirm everything is working smoothly again and this should not pose any problems for mainstream users as they will never update to the Oreo Beta in the first place.

I suppose you can mark this as fixed, then.

**Comment by The-Compiler** [comment]
@Apolyxon That sounds Samsung specific. I still see this issue on my Nokia 8 with the 8.0 release and February security patch level. The only place where it's been fixed upstream is 8.1 (or probably some feature patch on top of 8.0?) but 8.0 is not a beta.

**Comment by Apolyxon** [comment]
@The-Compiler 

Maybe Samsung cherry-picked the patch for their 8.0 release. Probs to them, if that is the case.

8.0 is not a beta but the problem was in the Samsung 8.0 Beta that I participated in. So I stand corrected: The mainstream SAMSUNG user will never experience this problem. Other manufacturers apparently still have this problem in their official 8.0 release.

**Comment by Querschlag** [comment]
Still experiencing both slow rendering when panning and zooming.
Nexus 6P running Android 8.1.0, patch level: February 5, 2018.
OsmAnd+ version 2.9.3 from Google Play

**Comment by Apolyxon** [comment]
@Querschlag 
Can you provide a gif/video of this behavior with Android 8.1?

**Comment by The-Compiler** [comment]
Hah - a few hours later, and I just got the Android 8.1 update on my Nokia 8 😆 With that, things are finally back to normal again! 👍

**Comment by Querschlag** [comment]
@Apolyxon 
I attached a screen recording.
![2018-02-15 19_29_30](https://user-images.githubusercontent.com/1651468/36273982-9c1488e4-1286-11e8-83a4-db8d212a775d.gif)

**Comment by vuurbeving** [comment]
We had similar problems in our app in 8.0. Our app does drawing on a canvas retrieved from a suerfacholder. I was able to work around it by using hardware accelerated drawing on a canvas, using https://developer.android.com/reference/android/view/SurfaceHolder.html#lockHardwareCanvas() , which is new in API Level 26, instead of `holder.lockCanvas()`. Quick search trough this osmand repo shows calls to `lockCanvas()`, but I'm completely unfamiliar with the osmand code so i'm not at all sure this is related to the issue or would improve anything at all.

Just posting this here since trough this issue I found the google issues about the drawBitmap bugs, which we definately were experiencing but went away after switching to a hardware accelerated canvas.

(might also be worth looking into https://developer.android.com/reference/android/view/Surface.html#lockHardwareCanvas() that's on the actual Surface which can be retrieved trought the surfaceholder **and** is available since API Level 23)

**Comment by vshcherb** [comment]
@vuurbeving Thanks for this hint! I think we will try to switch to it and give it a try possibly it will speedup the rendering.

**Comment by vshcherb** [comment]
I just quickly checked and the possible problems will be with
setPathEffect
setShadowLayer - probably not used any more

**Comment by tazva** [comment]
I see this thread, and several others, talking about performance issues. In my mind, this goes back to #3898, where something in OSM gets stuck eating up processor time, and makes everything it does, very slow and inefficient.

**Comment by ADepic** [comment]
I have zoom fps issues with a oneplus 2/running lineage os 15.1. While I can use osmand the zooming is insanely slow and is 2 - 15 fps.

HOWEVER this is not a problem of a one plus I have running oxygen is with android oreo, where the zooming is 30 - 60fps . 

So I can confirm this isn't a problem with oxygen os, but it is a problem with lineage os

**Comment by nicky-one** [comment]
The same issues on 2 devices with Oreo 8.1
- phone "wileyfox swift", arm64, crDroid 4.1
- tablet "sony xperia z2", arm, ResurrectionRemix 8.1

OSM 2.9.3 & latest nightly 8.4.2018 simply crashes and opens again:
- when tapping settings/dashboard (it does not open)
- during navigation: when trying to open the navigation settings

**Comment by contrapunctus-1** [comment]
I'm running OsmAnd v2.9.3 from F-Droid on a Nokia 6.1 (Android v8.1) and find that OsmAnd is significantly (though I wouldn't say unusably) slower on this device compared to a Moto G 5s Plus (Android v7.1.1), Samsung J7 Prime (Android v7.0), and OnePlus 3T (Android v8.0).

**Comment by Querschlag** [comment]
> So I can confirm this isn't a problem with oxygen os, but it is a problem with lineage os
It happens with Nexus and Pixel devices running Android Oreo and up as well. Just gets a bit less annoying with increasing capability of the SoC.

**Comment by bertin0** [comment]
I have some lag on a oneplus 3 on nitrogen os (based on AOSP 8.1) but it's not as bad as the ones posted above

**Comment by bertin0** [comment]
I have some lag on a oneplus 3 on nitrogen os (based on AOSP 8.1) but it's not as bad as the ones posted above

**Comment by bertin0** [comment]
Any idea what changes have to be made (to a rom) to fix this issue (like if you run a custom build?)

**Comment by tazva** [comment]
Run a CPU monitor which can log CPU/resource usage to a file (and/or on screen), I almost guarantee you will see OSM using the majority of CPU resources, which then slows down output. Regardless of ROM or version. This has been demonstrated & observed.

**Comment by zhangsr** [comment]
Looks like the fix : https://skia-review.googlesource.com/?polygerrit=0#/c/skia/+/56400/
which speed up skia drawing 32bit images into 565, and merged to android 8.1.0_release1

**Comment by ADepic** [comment]
This issue still isn't fixed for me - I am running android Oreo 8.1.0 Lineage os 15.1 on my oneplus 2.

Zooming in is 2 FPS!!!

**Comment by ADepic** [comment]
Zhangsr, what do you mean by Oreo 8.1.0_release1??

Do you mean the beta release that came before the stable release (because it should be fixed for me then) or a beta release that still is not stable yet?

**Comment by alxxnder** [comment]
Sorry for being so lame in my post, I am not Android devrloper, but there are other apps that work with OSM. For example, MAPS. ME and drawing speed is simply perfect, same as on Google maps. What is the difference in graphical output between these apps and Osmand? Other graphic library? Osmand was quite slow always, but since Android 8 it is unusable. As for me - it is the best navigation app, please please please fix it...

**Comment by ADepic** [comment]
+alxxnder
Maps.me uses a proprietary compression engine . Google maps does the same. That is why both the openstreetmap website and osmand have not instantaneous rendering.

**Comment by ADepic** [comment]
Why is this issue not in the future milestone?

**Comment by sonora** [comment]
@alxxnder Please do not confuse apps or web pages  with predefined maps with how we do it when you use our OsmAnd offline maps: With predefined (tile) maps, which are also supported in OsmAn, by the way, their appearance, content, degree of detail etc. is predetermined and fixed, these apps merely 'display the picture' at the section chosen and for the scale the user selects. OsmAnd's standard maps are 'vectorized' meaning they are compliled at display time from the underlying database, using all parameters and style sheets specified at that tine. While this is extremely more flexible in terms of e.g day vs. night appearance, degree of detail, etc., it is also slower.

@ADepic We are all researching what more can be done, at present it seems that all remaning possible fixes are Android finally being patched, which mostly (but apparently not for all devices?) happened in the 8.1 push.

**Comment by ADepic** [comment]
@sonora I have to mention thought online tiles are also extremely slow with osmand.
I think this - along with the super confusing UI (too many features) is what is getting you bad reviews on Google Play and Amazon.

**Comment by Querschlag** [comment]
@ADepic I don't think compression would help with this since the data is already present on device with offline maps. What could be optimized is (maybe) the query to the database and filtering or the drawing. As stated above graphic calls are obviously an issue.
The problem is that it seems nobody either received the patch or can report any improvements.
I tested on a Nexus 6P with latest updates and on a Google Pixel (1st. Gen.) and both have this issue.
Is there anyone with a 8.0/8.1 device not having these problems?

**Comment by ADepic** [comment]
@Querschlag "since the data is already present on device with offline maps. "
What do you mean by that? Because maps.me is also offline and it uses compression.

**Comment by ADepic** [comment]
@Querschlag 
@The-Compiler Apparently doesn't have this problem:
"Hah - a few hours later, and I just got the Android 8.1 update on my Nokia 8 laughing With that, things are finally back to normal again! +1"

**Comment by ADepic** [comment]
@Querschlag 
Another thing: This doesn't seem to apply to any Oneplus 3/3t running oxygen os. Maybe oxygen os somehow doesn't have this issue.

Evidence: My dad's phone is oneplus 3 on oreo 8 oxygen os 5 and hasn't got this error.
@raindropsfromsky doesn't seem to have this issue and he has oneplus 3t with oxygen os 5 (oreo 8.0)

**Comment by Germano0** [comment]
Confirming the OsmAnd 3.0 slowness on SoC Snapdragon 636 and Android 8.1

**Comment by Rovanion** [comment]
This is still an issue on a Sony Xperia XZ1 Compact with Android 8.0.0.

**Comment by ADepic** [comment]
Any progress? This still exists, and is the worst bug in osmand

**Comment by JulienPalard** [comment]
Fan confirm the issue on Nexus 6P with LineageOS 15.1-20180618-NIGHTLY-angler (Android 8.1.0).

**Comment by Digory123** [comment]
Got the issue with my Moto Z Play when Android 8.0.0 was released for the reteu channel a few days ago. I would agree that the main problem is zooming, but in my opinion the main menu (the one that can be swiped from the left) is also affected. Although I wouldn't say that Osmand is unusable since then, it is quite annoying...

**Comment by dvd73** [comment]
Honor 8 Lite 
Model: PRA-TL10
Build: PRA-TL10 8.0.0.335(C01)

After update from Android 7 to 8 got such issue.

**Comment by sedot42** [comment]
I'm also affected with a Sony Xperia X Compact with Android 8.0.0. It is very annoying and seems to also drain more battery, which is pretty bad when you're out hiking. As a paying customer I'm asking, why is this not a priority? You know about this bug for almost a year yet push features like wikivoyage integration while your core functionality, smoothly displaying a map, is almost broken for so many users. This leaves me scratching my head.

**Comment by The-Compiler** [comment]
Like people mentioned before multiple times, this is a bug in Android. If there's no good workaround for Osmand, there's little it can do, and you should probably go blame your phone manufacturer for not keeping you up to date.

**Comment by ADepic** [comment]
@The-Compiler

Not a single movibe mapplication that I know of suffers from any sort of lag at all. From Maps.Me , to Maps (100% FOSS fork of maps.Mr), to magic earth, to apple maps to google maps to Waze to HereWeGo they all have instantaneos zooming. BTW "Maps" (on fdroid) is 100% open source.

Why did Maps.Me not suffer from this??

**Comment by The-Compiler** [comment]
@ADepic Probably because they use prerendered (tiled) maps rather than vectorized ones - see e.g. 
the comments around https://github.com/osmandapp/Osmand/issues/4323#issuecomment-394178605 here.

**Comment by ADepic** [comment]
@The-Compiler

Maps.Me literally can't be raster. it has more compact (as in storage space ) maps, 40 mb world map compared to osmand 235 mb, 3d buildings, clickable POIs, night mode.

It literally has to be vector.

**Comment by ADepic** [comment]
Also, Sonora claims that rastor is faster, yet rastor tiles on openstreetmap.org aren't fast at all. Not to mention osmand rastor tiles are not instantaneous.

**Comment by bertin0** [comment]
@The-Compiler hmm... Even running a custom rom with all security patches up to date is still not good enough?

**Comment by JulienPalard** [comment]
I'm with @bertin0 on the question, I'm running Android 8.1.0 (LineageOS 15.1-20180702-NIGHTLY-angler), I can't "go blame your phone manufacturer for not keeping me up to date", and I have the issue.

Can it be related to [https://jira.lineageos.org/browse/BUGBASH-384](https://jira.lineageos.org/browse/BUGBASH-384)?

**Comment by The-Compiler** [comment]
@bertin0 @JulienPalard That's odd then. If you're running Android 8.1 I would expect this to be fixed - it was for me after I got the 8.1 upgrade FWIW.

**Comment by bertin0** [comment]
Is there no way to do the rendering on the GPU? Maybe that would be faster? (I don't really know what I'm talk g about xD)

**Comment by erikrr** [comment]
On my OnePlus i can force gpu rendering instead of cpu. Dont notice any
difference.
But after going to Android 8.1 it was solved.

A list with working mobiles and a list with not working mobiles would be
nice then people know what mobile then can buy and what not.



Op vr 6 jul. 2018 12:31 schreef bertin0 <notifications@github.com>:

**Comment by Rovanion** [comment]
Though such a list may be useful for debugging purposes for the developers
of Osmand I think it is quite unreasonable to think that people will chose
to not buy certain phones based off of a list of phones not "compatible"
with Osmand. The far more likely outcome is that people will only realise
that Osmand doesn't work well with their phone after the fact and then
simply replace Osmand with some other map app.

2018-07-06 12:49 GMT+02:00 erikrr <notifications@github.com>:

**Comment by JulienPalard** [comment]
The list won't solve the problem of people having already paid Osmand *before* the issue.

If of any help, I'm with my Nexus 6P around Paris, France. I'm not willing to solve the issue alone, but if an Osmand contributor is around, I will glady pay a beer [1] while experimenting with my phone to understand the issue.

[1] I can also offer non-alcoolic bevrages, let us be inclusive!

**Comment by erikrr** [comment]
True but for people like me who dont like other programs. For me OSMAND is
perfect when it works.
When i buy a new mobil i can look up the phone or it  works.

But the best thing would be if it just works.

I read something a few month back about a new rendering method somebody
suggested. I thought the maker would look at it. But never heard anything
about it.


Op vr 6 jul. 2018 12:55 schreef Rovanion Luckey <notifications@github.com>:

**Comment by utack** [comment]
>The list won't solve the problem of people having already paid Osmand before the issue.  
  
People also bought Software that worked on Windows 7, and not all of it works on Windows 10  
It is frustrating that an Android bug took down Osmand performance, but you can't expect developers to singlehandedly solve some problem that appears when the rug gets pulled out from under their feet.

**Comment by bertin0** [comment]
@utack i agree. I'm just trying to understand what could be done. I hope that on Android 9 it will be fixed for all devices

**Comment by aceman444** [comment]
On a Nokia 6.1 with Android 8.1 (Android One program, so the code should be up to date and unmodified coming from Google) with a midrange Snapdragon 630 CPU, the map rendering and panning is very fine - responding immediately.

But when zooming, there is noticeable stutter: just pressing zoom in or out there is about a 0.5s delay before any change. Then Osmand zooms by one step. And then there is nothing for about 1s before the missing areas at borders (when zooming out) are starting to be rendered. When panning to new areas, they are being re-rendered much faster - the rendering is not immediate and atomic, you can see roads and polygon layers to be appended, but at least it starts soon. But after zoom out, there is a delay where no rendering is done.

It seems, zoom in and out is implemented just by shrinking or multiplying the pixels in the current rendered map image so that is supposed to be fast (much faster than rerendering any area from the vector OSM data).

I don't say OsmAnd is unusable on this phone, it is great. I just say the zoom slowness may be mitigated, but still noticeable.

**Comment by xialady** [comment]
The map rendering is extremely slow on my Huawei P10 when zooming in and out, and when moving around, the map rendering is quite slow too. I am willing to help like @swedneck .

**Comment by Digory123** [comment]
I could observe that in my case the rendering speed depends on the area of the map. It is slower while my current position (the blue dot/cycle) is visible on the screen and faster when I zoom map sections outside my current position. 
This affects also the main menu. When I hit the three bars the menu opens much smoother if my current position is not inside the visible map section.
I searched an option to turn of the compass to see if the performance is better without drawing the layer for the direction angle, but I couldn't find...

**Comment by sedot42** [comment]
FYI: disabling animations (general settings) helps a bit.

**Comment by erikrr** [comment]
Also lowering the extra details under details 7/7 helps a little bit

Op ma 23 jul. 2018 13:01 schreef Stefan E. <notifications@github.com>:

**Comment by Xon2** [comment]
I am also experiencing this problem on a HTC U Ultra with android oreo 8.0. Whether that device will ever be updated to 8.1 or 9.0,...who knows, but i have my doubts. And from skimming over this thread, 8.1 seems only to help a subset of paying osmand customers. With my device, its mainly the zooming thats horribly slow. 

Question; why can't that deprecated library reinstalled with a patched osmand build? Is that library incompatible with android 8.0 or is it simply missing? 

I hate to say it, but this might be the nail in the coffin for osmand.

**Comment by cserpell** [comment]
I am making a +1 here. 

OsmAnd+ 3.0.4, released: 2018-06-11 (paid version)
Xiaomi Mi A1, Android 8.1.0, security patch level July 1, 2018
Kernel: 3.18.71-prf-gebb0eb7 (gcc version 4.9.x 20150123 (prerelease) (GCC) ) builder@mi-server /#1 Thu Jul 12 20:45:23 WIB 2018

OsmAnd was working perfectly until last week, that I decided to apply last monthly Android update. I already had Android 8.1, so it was not a change in Android version, but the update that it said included security patches. The details of the update are in [this page](https://source.android.com/security/bulletin/2018-07-01). Probably is something related.

**Comment by lineage22** [comment]
If of any help, like to buy a coffee to the one solving this issue: https://www.bountysource.com/issues/48512266-osmand-is-slow-on-android-o-fixed-in-android

**Comment by vshcherb** [comment]
@cserpell looks like it is a regression between 8.0 and 8.1.

**Comment by ADepic** [comment]
@vshcherb Do you know the bug know?

Is it still a android issue?
If so can you please summerize the problem so that I can drop a jira ticket over at lineage is. (So it could be fixed for all lineage is 15.1 users )
Is there something osmand can do?
If so, why is this not in the 3.1 milestone?

BTW, thanks for giving this a milestone

**Comment by vshcherb** [comment]
@ADepic  Please read conversation, there are links 
https://issuetracker.google.com/issues/64882137
https://issuetracker.google.com/issues/70113293
Nothing could be done so far in OsmAnd itself. I plan to summarize that issue, so it will be easier to track which devices are affected.

**Comment by refacto** [comment]
Can confirm same very slow behaviour with a OnePlus One, running on LineageOS 15.1 `15.1-20180806-microG-bacon`. The phone also gets noticeable warmer when lots of tiles are rendered, much more than before.

**Comment by ldldr** [comment]
I can confirm lags when zooming on a Redmi Note 4 with lineage-15.1-20180807-microG-mido.

**Comment by darkomilicevic89** [comment]
I have the same problem with my Huawei p10 lite, android 8.0. It's better when you turn off Contour lines and animation, but still very sluggish.

**Comment by Digory123** [comment]
@darkomilicevic89 That's strange. On my daughter's Huawei P10 lite with Android 8.0 Osmand is running fine, at least much better than on my Moto Z Play with Android 8.0. Did you install a custom ROM or is it the official Android Oreo update for your device?

**Comment by darkomilicevic89** [comment]
I didn't install custom rom, it's official. Build number WAS-LX1 8.0.0.362(C738), Android security patch is from July 2018, that's the latest version for my phone. I forgot to mention that my OsmAnd version is 3.0.4 (from f-droid), and on google play store the latest is, I think 3.1.3. Maybe the app update will fix this problem...

**Comment by Digory123** [comment]
@darkomilicevic89 My daughter's P10 lite is build number WAS-LX1 8.0.0.366(C432).  So I would agree with sdever that you should wait for the next Android update...

**Comment by Digory123** [comment]
@vshcherb In https://issuetracker.google.com/issues/70113293 I read that the problem only occurs with Bitmap.Config RGB_565. Wouldn't it be possible to change it to ARGB_8888 for example?
As I'm not an Android developer, I hope you will forgive me this question...

**Comment by alxxnder** [comment]
After Osmand updated to 3.1.5 issue suddenly got lot better. Lag when zooming got SIGNIFICANTLY smaller. Panning is totally OK now. I think now it is like before on earlier versions of Android. Android version is still 8.0.0 and there was no update before the improvement. Device is Huawei mate 10.

**Comment by Digory123** [comment]
I cannot confirm that 3.1.5 improves the zooming behaviour. On my phone it is the same as before. Panning was ok all the time.

**Comment by erikrr** [comment]
Maybe switching to iPhone when buying a new phone, that always seems to
work. A pity the versions are not alike and Android is further in
development.
It feels strange and unsafe that each Google update could 'kill' the app.

Op zo 12 aug. 2018 19:08 schreef Digory123 <notifications@github.com>:

**Comment by cserpell** [comment]
This is exactly what I mean regarding "not our fault". Bugs like this drive
users, specially new users, away from OsmAnd.

**Comment by prototype99** [comment]
the simplest way to put it: 'not our fault' does not need to also mean 'this is okay'. outside of that, after the most recent update, panning is delayed about half a second, zooming is delayed about a second. whilst those are small delays... they are large compared to other available options and could be a large productivity impairment for some

**Comment by JulienPalard** [comment]
I cannot confirm that 3.1.5 improves the zooming behaviour: 3.1.5 is still not on f-droid :-1:  (Only having 3.0.4 here).

**Comment by cserpell** [comment]
I confirm that 3.1.5 is working better, at least in my device. Thanks all.

**Comment by vshcherb** [comment]
Issue is closed to be managed in a better way  in #5878
